pkg/datapath/loader/base.go

```diff
@@ -334,19 +363,28 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 	if option.Config.Tunnel == option.TunnelDisabled && option.Config.EnableIPv4EgressGateway {
 		// Enable tunnel mode to vxlan if egress gateway is configured
 		// Tunnel is required for egress traffic under this config
-		args[initArgTunnelMode] = option.TunnelVXLAN
+
+		// We are effectively changing configuration here, why isn't this reflected in the actual config value?
+		// Should these checks move to initEnv() in daemon/cmd/daemon_main.go?
+		// Changing config here feels "dangerous" as I don't know if we rely on those values being unchanged down the road.
+		// args[initArgTunnelMode] = option.TunnelVXLAN
+		option.Config.Tunnel = option.TunnelVXLAN
```

Comment:
Not sure if this is the right way to go as the comment above explains

Reply:
`option.Config.Tunnel` holds the routing mode (and the overlay protocol in case of tunneling). Using egress gateway doesn't mean we change the routing mode for all traffic. Only traffic going through the egress gateways is going to flow through an overlay. Everything else will remain unencapsulated if using native routing mode.

Note that this code changed with https://github.com/cilium/cilium/pull/24561, which separates the routing mode (`option.Config.RoutingMode`) from the tunneling protocol (`option.Config.TunnelProtocol`).

Reply:
Ok, I'll change it back to work on a separate variable and not on the config options directly. Thanks for the explanation :pray: